packages/image_picker/image_picker_platform_interface/lib/src/platform_interface/image_picker_platform.dart

```diff
@@ -168,6 +168,45 @@ abstract class ImagePickerPlatform extends PlatformInterface {
     throw UnimplementedError('pickVideo() has not been implemented.');
   }
 
+  /// Returns a [PickedFile] with the image that was picked.
+  ///
+  /// The `source` argument controls where the image comes from. This can
+  /// be either [ImageSource.camera] or [ImageSource.gallery].
+  /// 
+  /// The `type` argument controls what the content return type. This can
+  /// be either [RetrieveType.image] or [RetrieveType.video].
+  ///
+  /// If specified, the image will be at most `imageMaxWidth` wide and
+  /// `imageMaxHeight` tall. Otherwise the image will be returned at it's
+  /// original width and height.
+  ///
+  /// The `imageQuality` argument modifies the quality of the image, ranging from 0-100
+  /// where 100 is the original/max quality. If `imageQuality` is null, the image with
+  /// the original quality will be returned. Compression is only supported for certain
+  /// image types such as JPEG. If compression is not supported for the image that is picked,
+  /// an warning message will be logged.
+  /// 
+  /// The [videoMaxDuration] argument specifies the maximum duration of the captured video. If no [videoMaxDuration] is specified,
+  /// the maximum duration will be infinite.
+  ///
+  /// Use `preferredCameraDevice` to specify the camera to use when the `source` is [ImageSource.camera].
+  /// The `preferredCameraDevice` is ignored when `source` is [ImageSource.gallery]. It is also ignored if the chosen camera is not supported on the device.
+  /// Defaults to [CameraDevice.rear].
+  ///
+  /// In Android, the MainActivity can be destroyed for various reasons. If that happens, the result will be lost
+  /// in this call. You can then call [retrieveLostData] when your app relaunches to retrieve the lost data.
+  Future<PickedFile> pickMedia({
+    @required ImageSource source,
+    @required RetrieveType type,
+    double imageMaxWidth,
+    double imageMaxHeight,
+    int imageQuality,
+    Duration videoMaxDuration,
+    CameraDevice preferredCameraDevice = CameraDevice.rear,
```

Comment:
I like this change, but I'd prefer to add this after the web version is in, to not add a new platform + new functionality. Can we do this in a follow up to the current change?

About the functionality itself, this seems to be a big fusion of all the pick* methods that are currently implemented. If "pickAudio" or "pickFile" get added, this method will grow in complexity (it'll probably need *more* parameters).

Maybe the best way to go is to have a completely agnostic "pickFile" method that lets you select *anything* from the disk.

That API can have the RetrieveType parameter if we want to limit to *some* types, but it wouldn't  concern itself with "imageQuality" or "videoDuration". 

That would be a much simpler API, if you needed to pick from multiple types IMO!